### PR TITLE
Intrepid2: parallelize the Lagragian Interpolation tool

### PR DESCRIPTION
**Pull request by mperego**
@trilinos/intrepid2 

## Description

Improve Lagrangian Interpolation code by doing a parallel for over the cells. Limiting now to one level of parallelism. 
Added OpenMP and Cuda tests (w/ UVM for now)

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [ ] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.
- [x ] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

File: packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp [hunk 1]
```diff
@@ -56,6 +56,217 @@
...
 namespace Experimental {
 
+
+
+template<typename scalarViewType,
+typename ortViewType,
+typename t2oViewType,
+typename subcellParamViewType,
+typename intViewType>
+struct F_computeDofCoordsAdCoeff {
```
Comment (CamelliaDPG) [inline packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp]: Typo?  Should be `F_computeDofCoordsAndCoeff`,  I think…

File: packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp [hunk 2]
```diff
@@ -56,6 +56,217 @@
...
+  const scalarViewType facesInternalDofCoords, faceDofCoeffs, ortJacobianFace, refFaceTangents, refFacesNormal;
+  scalarViewType edgeWorkView, faceWorkView;
+  const ordinal_type cellDim, numEdges, numFaces;
+  const ordinal_type edgeTopoKey, numEdgeInternalDofs;
+  const intViewType faceTopoKey, numFacesInternalDofs;
+  const value_type edgeScale, faceScale;
+  const bool isBasisHCURL, isBasisHDIV;
+
+  F_computeDofCoordsAdCoeff( scalarViewType dofCoords_,
+      scalarViewType dofCoeffs_,
```
Comment (CamelliaDPG) [inline packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp]: Isn't the usual convention to use trailing underscores for member variables?  You seem to have reversed this here, leaving member variables unadorned, and using trailing underscores in the constructor argument.

File: packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp (hunk 1 above)
Comment (mperego) [reply packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp]: oh, right, I'll fix it

File: packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp (hunk 2 above)
Comment (mperego) [reply packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp]: Not sure. It's the same in Intrepid2_OrientationToolsDefModifyBasis.hpp. But I can change it if it is better.

Comment (CamelliaDPG) [reply packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp]: It is at least a fairly common convention to do what I suggested (though whether underscores lead or trail varies).  I think it is better; it allows the reader to distinguish between member variables and variables that are local to a method at a glance.

Comment (mperego) [reply packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp]: OK

**Comment by mperego** [comment]
@CamelliaDPG I think the code can run UVM free. The tests need to be updated though. I'll do that once I've parallelized the other projection tools.

**Comment by mperego** [comment]
@CamelliaDPG  can you re-approve?

File: packages/intrepid2/src/Orientation/Intrepid2_OrientationToolsDefModifyPoints.hpp [hunk 3]
```diff
@@ -274,7 +274,7 @@ namespace Intrepid2 {
 
     template<typename outPointViewType,
              typename refPointViewType>
-    inline
+    KOKKOS_INLINE_FUNCTION
```
[the remark this answers is not in the document]
Comment (mperego) [reply packages/intrepid2/src/Orientation/Intrepid2_OrientationToolsDefModifyPoints.hpp]: It's an enum, isn't it fine? It works on GPUs tested by the PR testing.

**Pull request merged**